File: Java/labs/lab15_Command/task_2_1/src/Document.java
public class Document {

    private String name;

    Document(String name) {
        System.out.println("Creating new document " + name);
    }

    void save() {
        System.out.println("Saving document " + name + "...");
    }

    void showInfo() {
        System.out.println("Document " + this.name);
    }

    void print() {
        System.out.println("Printing document " + name + "...");
    }
}

File: Cpp/lab15_Command/task_2_1 - solution/src/document.cpp
#include <iostream>

#include "document.hpp"


Document::Document(std::string const& name)
: name(name)
{
  std::cout << "Creating new document " << this->name << std::endl;
}

void Document::save()
{
  std::cout << "Saving document " << this->name << "..." << std::endl;
}

void Document::showInfo() const
{
  std::cout << "Document " << this->name << std::endl;
}

void Document::print() const
{
  std::cout << "Printing document " << this->name << "..." << std::endl;
}
